### PR TITLE
Update Keras CV APIs for layers

### PR DESCRIPTION
**Pull request by sachinprasadhs**
Newly listed keras CV layers of

- object_detection
- object_detection_3d
- preprocessing
- regularization

Could not come up with the right category for `Window Partitioning layer`, commented it as of now.

Reorganized, Layers_master to have single nested structure, similar to Keras core.

### CODE REVIEW

File: scripts/cv_api_master.py [hunk 1]
```diff
@@ -1,127 +1,470 @@
...
+                },
+                {
+                    "path": "overlapping_patching",
+                    "title": "Overlapping Patching and Embedding layer",
+                    "generate": [
+                        "keras_cv.layers.OverlappingPatchingAndEmbedding"
+                    ]
+                },
+                {
+                    "path": "patching_embedding",
```
Comment (fchollet) [inline scripts/cv_api_master.py:273]: patching_and_embedding (must following the layer name)

File: scripts/cv_api_master.py [hunk 2]
```diff
@@ -1,127 +1,470 @@
...
+                },
+            ]
+        },
+        {
+            "path": "embedding/",
+            "title": "Embedding layers",
+            "toc": True,
+            "children": [
+                {
+                    "path": "relative_positional_embedding",
```
Comment (fchollet) [inline scripts/cv_api_master.py:252]: add_relative_positional_embedding

File: scripts/cv_api_master.py [hunk 3]
```diff
@@ -1,127 +1,470 @@
...
+                    "title": "Add Relative Positional Embedding layer",
+                    "generate": [
+                        "keras_cv.layers.AddRelativePositionalEmbedding"
+                    ]
+                },
+                {
+                    "path": "multihead_relative_pe",
```
Comment (fchollet) [inline scripts/cv_api_master.py:259]: multi_head_attention_with_relative_pe

File: scripts/cv_api_master.py [hunk 4]
```diff
@@ -1,127 +1,470 @@
...
+                    "title": "Vectorized BaseImage Augmentation layer",
+                    "generate": ["keras_cv.layers.VectorizedBaseImageAugmentationLayer"],
+                },
+            ],
         },
         {
-            "path": "equalization",
-            "title": "Equalization layer",
-            "generate": ["keras_cv.layers.Equalization"],
+            "path": "convolution/",
```
Comment (fchollet) [inline scripts/cv_api_master.py:209]: Please create a separate category "modeling" and move the conv layers and the encoder layers there.

File: scripts/cv_api_master.py [hunk 5]
```diff
@@ -1,127 +1,470 @@
...
+                    "title": "Region of interests (ROI) Pooler",
+                    "generate": [
+                        "keras_cv.layers.ROIPooler"
+                    ]
+                },        
+            ]
+        },
+        {
+            "path": "object_detection_3d/",
+            "title": "Object Detection 3D layers",
```
Comment (fchollet) [inline scripts/cv_api_master.py:406]: 3D object detection layers

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: scripts/cv_api_master.py [hunk 6]
```diff
@@ -1,127 +1,477 @@
...
+                    "path": "window_partitioning",
+                    "title": "Window Partitioning layer",
+                    "generate": [
+                        "keras_cv.layers.WindowPartitioning"
+                    ]
+                },
+            ]
+        },
+        {
+            "path": "modelling/",
```
Comment (fchollet) [inline scripts/cv_api_master.py:275]: Let's go with the US spelling, `modeling`.

File: scripts/cv_api_master.py [hunk 7]
```diff
@@ -1,127 +1,477 @@
...
+                    "title": "Vectorized BaseImage Augmentation layer",
+                    "generate": ["keras_cv.layers.VectorizedBaseImageAugmentationLayer"],
+                },
+            ],
         },
         {
-            "path": "equalization",
-            "title": "Equalization layer",
-            "generate": ["keras_cv.layers.Equalization"],
+            "path": "decoder/",
```
Comment (fchollet) [inline scripts/cv_api_master.py:209]: Remove this section, it is redundant with `modeling`.

File: scripts/cv_api_master.py [hunk 8]
```diff
@@ -1,127 +1,477 @@
...
+                        "keras_cv.layers.WindowPartitioning"
+                    ]
+                },
+            ]
+        },
+        {
+            "path": "modelling/",
+            "title": "Modelling layers",
+            "toc": True,
+            "children": [
```
Comment (fchollet) [inline scripts/cv_api_master.py:278]: You can remove the sub-section and just put all the layers directly under `modeling`.

**Review by fchollet — COMMENTED** [review]
Thanks for the updates!

File: scripts/cv_api_master.py (hunk 7 above)
Comment (sachinprasadhs) [reply scripts/cv_api_master.py:209]: Where do I keep the `FeaturePyramid` and `SpatialPyramidPooling` layers.

File: scripts/cv_api_master.py (hunk 8 above)
Comment (sachinprasadhs) [reply scripts/cv_api_master.py:278]: So will it be something like `MODELING_MASTER`?

**Pull request closed without merging**